Sources/MarathonCore/ArgumentParser.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  ArgumentParser.swift
+//  Marathon
+//
+//  Created by pixyzehn on 2017/04/09.
+//
+//
+
+import Foundation
+
+public struct ArgumentParser: CustomStringConvertible {
+    private var arguments: [ArgumentType]
+
+    private enum ArgumentType: CustomStringConvertible {
+        case argument(String)
+        case option(String)
```

Comment:
I think it is a good idea to make it simple, but SwiftPM uses options for `--list` like this. How about using `Option` struct instead of `Flag` struct? 

```sh
OPTIONS:
  --build-path            Specify build/cache directory [default: ./.build]
  --chdir, -C             Change working directory before any other operation
  --color                 Specify color mode (auto|always|never) [default: auto]
  --enable-prefetching    Enable prefetching in resolver
  --verbose, -v           Increase verbosity of informational output
  -Xcc                    Pass flag through to all C compiler invocations
  -Xlinker                Pass flag through to all linker invocations
  -Xswiftc                Pass flag through to all Swift compiler invocations
  --help                  Display available options
```

In this case, `Option` struct has `long` and `short` member.

```swift
struct Option {
    let long: String
    let short: String?

    init(long: String, short: String? = nil) {
        self.long = long
        self.short = short
    }

    func contains(option: Option) -> Bool {
        if let short = short {
            return long == option.long || short == option.short
        } else {
            return long == option.long
        }
    }
}

let option = Option(long: "--list", short: "-l")
let option2 = Option(long: "--no-xcode")
```